app/src/main/kotlin/org/fossify/filemanager/activities/SaveAsActivity.kt

```diff
@@ -15,10 +15,24 @@ import java.io.File
 class SaveAsActivity : SimpleActivity() {
     private val binding by viewBinding(ActivitySaveAsBinding::inflate)
 
-    override fun onCreate(savedInstanceState: Bundle?) {
+
+        override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
+        tryInitFileManager()
+    }
 
+    private fun tryInitFileManager() {
+        handleStoragePermission { granted ->
+            if (granted) {
+                saveAsDialog()
+            } else {
+                toast(R.string.no_storage_permissions)
+                finish()
+            }
+        }
+    }
+    private fun saveAsDialog() {
```

Comment:
Missing an empty line here.

```suggestion
    }

    private fun saveAsDialog() {
```